src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -311,8 +311,8 @@ protected Marshaller()
             }
             else
             {
-                // Passing as [Out] is valid only if it is not ValueType nor string
-                if (!parameterType.IsValueType && !parameterType.IsString)
+                // Passing as [Out] is valid only if it is not ValueType, string nor pointer
+                if (!parameterType.IsValueType && !parameterType.IsString && !parameterType.IsPointer)
```

Comment:
Nit: I haven't actually checked, but `&& !parameterType.IsFunctionPointer` should probably be handled the same.